### PR TITLE
fix(reservation): update reservation on volume resize

### PR DESCRIPTION
**Pull request by sinhaashish**
## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**:

**What this PR does?**:

**Does this PR require any upgrade changes?**:

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/website is used to track them:

**PLEASE REMOVE BELOW INFORMATION BEFORE SUBMITTING**

The PR title message must follow convention:
   `<type>(<scope>): <subject>`.

Where: <br />
- `type` is defining if release will be triggering after merging submitted changes, details in [CONTRIBUTING.md](../CONTRIBUTING.md).
    Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

- `scope` is a single word that best describes where the changes fit.
  Most common scopes are like:
    - data engine (`localpv`, `jiva`, `cstor`)
    - feature (`provisioning`, `backup`, `restore`, `exporter`)
    - code component (`api`, `webhook`, `cast`, `upgrade`)
    - test (`tests`, `bdd`)
    - chores (`version`, `build`, `log`, `travis`)

- `subject` is a single line brief description of the changes made in the pull request.

### CODE REVIEW

**Comment by sinhaashish** [comment]
Fixes #568

File: pkg/zfs/zfs_util.go [hunk 1]
```diff
@@ -299,6 +299,11 @@ func buildVolumeResizeArgs(vol *apis.ZFSVolume) []string {
 		ZFSVolArg = append(ZFSVolArg, volsizeProperty)
 	}
 
+	if vol.Spec.ThinProvision == "no" {
+		reservationProperty := "reservation=" + vol.Spec.Capacity
+		ZFSVolArg = append(ZFSVolArg, "-o", reservationProperty)
+	}
```
Comment (Abhinandan-Purkait) [inline pkg/zfs/zfs_util.go:305]: I believe after @cinapm's PR , we need to set reservation or refreservation based on the quota type? Also we need a change in line 295.

@cinapm Any thoughts here?

**Comment by abuisine** [comment]
I implemented QuotaType support here #595

**Comment by sinhaashish** [comment]
closing this as its addressed on PR https://github.com/openebs/zfs-localpv/pull/595

**Pull request closed without merging**